packages/core/src/oidc/init.ts

```diff
@@ -344,5 +351,23 @@ export default function initOidc(
   });
   oidc.use(koaBodyEtag());
 
+  oidc.use(async (ctx, next) => {
+    await next();
+
+    // https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#registering-module-middlewares-helmet-ip-filters-rate-limiters-etc
+    if (ctx.oidc?.route === 'token') {
+      const userIdResult = z.string().safeParse(ctx.oidc.entities.Account?.accountId);
+
+      if (userIdResult.success) {
+        // Mark this user as active today
+        await insertActiveUser({
+          id: generateStandardId(),
+          userId: userIdResult.data,
+          date: getUtcStartOfToday().getTime(),
+        });
```

Comment:
Not sure, but do we need to add an extra layer of try cache here? 
If this SQL throws exceptions, should we interrupt the whole API? Especially on a token exchange flow, OIDC may successfully issue a new token but the end user won't receive it.

Also how about using the [oidc event listeners](https://github.com/panva/node-oidc-provider/blob/main/docs/events.md#events)?  Might reduce the duplicate insert call, as it can be attached to some authorization event exclusively?

Reply:
Event listener is cool, updated